src/pem_common_plug.c

```diff
@@ -51,6 +54,16 @@ int pem_valid(char *ciphertext, struct fmt_main *self)
 	ctcopy += TAG_LENGTH;
 	if ((p = strtokm(ctcopy, "$")) == NULL) // type
 		goto err;
+	if (*p != '1' && !warned) {
+		warned = 1;
+		if (strcmp((p = strtokm(NULL, "$")), "pbkdf2") != 0) {
+			fprintf(stderr, "Warning: " FORMAT_LABEL " kdf algorithm <%s> is not supported currently!\n", p);
+			goto err;
+		}
+		p = strtokm(NULL, "$");
+		fprintf(stderr, "Warning: " FORMAT_LABEL " prf algorithm <%s> is not supported currently!\n", p);
+		goto err;
```

Comment:
In these two newly added places, you also need to check the return value from `strtokm` for (non-)NULL before using it, like this function does elsewhere.

Reply:
Thanks. A question, when it comes to prf algorithms in `john`, are they always variants of `sha*`?

Reply:
I will push a new PR, happy to know whether my assumptions are correct there.

Reply:
> A question, when it comes to prf algorithms in john, are they always variants of sha*?

We shouldn't be making that assumption, and besides we're talking about what could be used in future PEM-encoded data not supported "in john". So I suggest you drop that `strstr` part.

I guess/hope that by "new PR" you merely meant another force-push to this one.

Reply:
yeah, that's what I meant, wrong wording.